docs/rules/data-testid.md

```diff
@@ -0,0 +1,46 @@
+# Enforces consistent naming for the data-testid attribute (data-testid)
+
+Ensure `data-testid` values match a provided regex. This rule is un-opinionated, and requires configuration.
+
+## Rule Details
+
+> Assuming the rule has been configured with the following regex: `^TestId(\_\_[A-Z]*)?$`
+
+Examples of **incorrect** code for this rule:
+
+```js
+const foo = props => <div data-testid="my-test-id">...</div>;
+const foo = props => <div data-testid="myTestId">...</div>;
+const foo = props => <div data-testid="TestIdEXAMPLE">...</div>;
+```
+
+Examples of **correct** code for this rule:
+
+```js
+const foo = props => <div data-testid="TestId__EXAMPLE">...</div>;
+
+const bar = props => <div data-testid="TestId">...</div>;
+
+const baz = props => <div>...</div>;
+```
+
+## Options
+
+| Option        | Details                                                                                                                                                                                                                                                           | Example                                       |
+| ------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | --------------------------------------------- |
+| testIdPattern | A regex used to validate the format of the `data-testid` value. `{componentName}` can optionally be used as a placeholder and will be substituted with the name of the file OR the name of the files parent directory in the case when the fileName is `index.js` | `'^{componentName}(\_\_([A-Z]+[a-z]_?)+)_\$'` |
+| excludePaths  | An array of path strings to exclude from the check                                                                                                                                                                                                                | `["__tests__"]`                               |
```

Comment:
I'm wondering if this is really necessary, as it can be achieved with the eslint config `overrides`. Something like this I suppose:

```js
{
  "rules": {
    
  },
  "overrides": [
    {
      "files": ["*.js"],
      "ignorePatterns": "__tests__",
      "rules": {
        "testing-library/consistent-data-testid": [
          2,
          { "testIdPattern": "^TestId(__[A-Z]*)?$" }
        ]
      }
    }
  ]
}
```